rfcs/0000-policy_target_scope.md

```diff
@@ -0,0 +1,228 @@
+# RFC A single Policy scoped to the whole targeted HTTPRoute
+
+- Feature Name: Policy scoped to the whole targeted Route
+- Start Date: 2022-11-09
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Currently both `RateLimitPolicy` and `AuthPolicy` allow for a policy to only affect a _subset_ of the
+traffic on a targeted `HTTPRoute`. This proposal aims at simplifying the scope of a policy to target
+_all_ the traffic of a route.
+
+# Motivation
+[motivation]: #motivation
+
+As per the specification, a given resource will always need to support multiple policies targeting it.
+Even if we only support a single e.g. `RateLimitPolicy` per resource, there will alway be a
+possibility for another policy being defined else where in that resource's hierarchy: e.g. _rlp1_
+targeting the `Gateway`, while _rlp2_ targeting the `HTTPRoute` being attached to the `Gateway`. Until
+there is an `HTTPRoute` tho, _rlp1_ will have no semantic, as there is no traffic to rate limit. When
+an `HTTPRoute` is attached to the `Gateway` is the point when that _rlp1_ becomes applicable. If that
+`HTTPRoute` also defines a `RateLimitPolicy`, the two get merged, according to the `default` &
+`override` rules of the specification. 
+
+The idea is to keep things simple (if only for now), so that when traffic is routed through an
+`HTTPRoute` all applicable policies of a `kind` have been merged into a single one _and_ that single
+policy then applies to the whole traffic of that route.
+
+The current approach, which lets each policy define an arbitrary subset of the traffic to which it
+applies, makes it impossible to know how to merge the different policies into one. Each policy's set
+requires to know how to applicable it becomes to the union and/or the intersection of the said
+subsets. While a heuristic could be applied, there are use-cases where the other option will be
+desirable. The question then becomes which policy is the authority as to decide which it should be. In
+order to avoid having to answer these questions, this proposal looks at not letting a policy
+discriminate which part of traffic it applies to: it's the whole traffic, always.
```

Comment:
Ok so in this case it would be up to me to define the auth or rate limits based on the path/method as part of defining the limits within the policy or the authschema? So if I wanted no limits on a particular endpoint or no auth on an endpoint

Reply:
ok reading on further I see that is it is a much more limited API. I can see how this will help with overrides and defaults, but my concern is that it is too limited to really be useful?